force-app/main/default/classes/CON_DeleteContactOverride_TEST.cls

```diff
@@ -551,4 +640,82 @@ private class CON_DeleteContactOverride_TEST {
+
+        private Id contactId;
+        private Boolean canDeleteContact;
+        private Id accountId;
+        private Boolean canDeleteAccount;
+        private Boolean isAccountWithSingleContact;
+
+        final public String CONTACT_NAME = 'Test Contact';
+        final public String ACCOUNT_NAME = 'Test Account';
+
+        CON_DeleteContactOverrideStub(Id contactId, Boolean canDeleteContact, Id accountId, Boolean canDeleteAccount,
+            Boolean isHouseholdWithSingleContact)
+        {
+            this.contactId = contactId;
+            this.canDeleteContact = canDeleteContact;
+            this.accountId = accountId;
+            this.canDeleteAccount = canDeleteAccount;
+            this.isAccountWithSingleContact = isHouseholdWithSingleContact;
+        }
+
+        private Contact queryContact() {
+            Contact contactDetails = (Contact) JSON.deserialize('{' +
+                '"Id": "' + this.contactId + '", ' +
+                '"Name": "' + CONTACT_NAME + '", ' +
+                '"FirstName": "' + 'Test' + '", ' +
+                '"LastName": "' + 'Contact' + '", ' +
+                '"AccountId": "' + this.accountId + '", ' +
+                '"Account": {"Name": "' + ACCOUNT_NAME + '", ' +
+                '"npe01__SYSTEMIsIndividual__c": true}' +
+                '}', Contact.class);
+            return contactDetails;
+        }
+
+        public Object handleMethodCall(
+            Object stubbedObject, String stubbedMethodName, Type returnType,
+            List<Type> listOfParamTypes, List<String> listOfParamNames, List<Object> listOfArgs
+        ) {
+            switch on stubbedMethodName {
+                when 'canDeleteContactRecord' {
+                    return this.canDeleteContact;
+                }
+                when 'canDeleteAccountRecord' {
+                    return this.canDeleteAccount;
+                }
+                when 'queryContact' {
+                    return this.queryContact();
+                }
+                when 'isAccountWithSingleContact' {
+                    return this.isAccountWithSingleContact;
+                }
+            }
+            return null;
+        }
+    }
+
+    /**
+     * Credit to fflib-apex-mocks fflib_IDGenerator
+     * https://github.com/apex-enterprise-patterns/fflib-apex-mocks/blob/master/sfdx-source/apex-mocks/main/classes/fflib_IDGenerator.cls
+     */
+
+    private class IDGenerator {
+        private Integer fakeIdCount = 0;
+        private final String ID_PATTERN = '000000000000';
+
+
+        private Id generate(Schema.SObjectType sObjectType) {
+            String keyPrefix = UTIL_Describe.getObjectDescribe(sObjectType).getKeyPrefix();
+            fakeIdCount++;
+
+            String fakeIdPrefix = ID_PATTERN.substring(0, ID_PATTERN.length() - String.valueOf(fakeIdCount).length());
+
+            return Id.valueOf(keyPrefix + fakeIdPrefix + fakeIdCount);
+        }
+    }
```

Comment:
Let's pop this out into its own class file. Once that's done, we should be able to delete the one in RelationshipsTreeGrid_TEST.cls too. Eventually, we may replace it with the fflib Id Generator class.